src/webR/robj.ts

```diff
@@ -767,6 +825,22 @@ abstract class RObjAtomicVector<T extends atomicType> extends RObjImpl {
 }
 
 export class RObjLogical extends RObjAtomicVector<boolean> {
+  constructor(val: RTargetObj | RObjAtomicData<boolean>) {
+    if (isRTargetObj(val)) {
+      super(val);
+      return this;
+    }
+    const values = isRObjectTree(val) ? val.values : Array.isArray(val) ? val : [val];
+    const ptr = Module._Rf_protect(Module._Rf_allocVector(RTypeMap.logical, values.length));
+    values.forEach((v, i) =>
+      Module.setValue(Module._LOGICAL(ptr) + 4 * i, v === null ? RObjImpl.naInt : Boolean(v), 'i32')
```

Comment:
Should probably use an lgl NA alias even if the underlying representation is int NA. This is what the C API of R does.

```suggestion
      Module.setValue(Module._LOGICAL(ptr) + 4 * i, v === null ? RObjImpl.naLgl : Boolean(v), 'i32')
```

Reply:
Thanks! I'll pull out `LOGICAL()` and the other vector block data pointer lookups into a separate variable outside the loop.